content/news/2020-06-19-gsoc-update2-2020.md

```diff
@@ -0,0 +1,15 @@
++++
+title = "Oops, Now Revert"
+date = 2020-06-19
+extra.author = "Devin Ragotzy"
++++
+
+This week in ruma-events' Google Summer of Code project, after trying out the new API in [`matrix-rust-sdk`](https://github.com/DevinR528/matrix-rust-sdk/pull/2), we found a few fatal flaws. There was no easy way to go from `StateEvent<AnyStateEventContent>` to `StateEvent<SpecificEventContent>`, the other issue was one could create a `StateEvent` with differing `content` and `prev_content` fields using the `AnyStateEventContent` enum. The [new API](https://github.com/ruma/ruma/pull/68) will be similar to the old one, each event kind (state, message, ephemeral, etc.) is represented by an enum.
```

Comment:
"the new API" could easily be interpreted as all your previous work here I think. Maybe make it clear that it's only some parts that need to be changed back ;)

Reply:
> This week in ruma-events' Google Summer of Code project, after trying out the new `Any*Event` enums [`matrix-rust-sdk`](https://github.com/DevinR528/matrix-rust-sdk/pull/2), we found a few fatal flaws. There was no easy way to go from `StateEvent<AnyStateEventContent>` to `StateEvent<SpecificEventContent>`, the other issue was one could create a `StateEvent` with differing `content` and `prev_content` fields using the `AnyStateEventContent` enum. The [new API](https://github.com/ruma/ruma/pull/68) will be similar to the old one, each event kind (state, message, ephemeral, etc.) is still a generic struct but, each event type (think "m.room.aliases") will be represented by an enum instead of the `content` field keeping track of the event type.

Reply:
"new" and "old" API are still used here and probably not very clear. Also, maybe say "big" flaws? ^^